components/org.wso2.carbon.identity.oauth.ui/src/main/resources/web/oauth/add-finish-ajaxprocessor.jsp

```diff
@@ -150,19 +150,30 @@
                     registeredScopeValidators.add(allowedValidator);
                 }
             }
-            
+
             if (OAuthConstants.OAuthVersions.VERSION_2.equals(oauthVersion)) {
                 app.setGrantTypes(grants);
                 app.setScopeValidators(registeredScopeValidators.toArray(new String[registeredScopeValidators.size()]));
             }
-            
-            if (Boolean.parseBoolean(request.getParameter("enableAudienceRestriction"))) {
-                String audiencesCountParameter = request.getParameter("audiencePropertyCounter");
-                if (IdentityUtil.isNotBlank(audiencesCountParameter)) {
-                    int audiencesCount = Integer.parseInt(audiencesCountParameter);
-                    String[] audiences = request.getParameterValues("audiencePropertyName");
+
+            if (Boolean.parseBoolean(request.getParameter("enableIdTokenAudienceRestriction"))) {
```

Comment:
better to define constants

Reply:
resolved with 9bd3f390ccf683a339e2b1cab3c1af84f86a7acb